api/views.py

```diff
@@ -50,6 +53,45 @@ def unauthorized(message="You must be logged in"):
     return JsonResponse({"statusCode": 401, "error_message": message}, status=401)
 
 
+@csrf_exempt
+def upload_image(request):
+    # import pdb; pdb.set_trace()
+    if request.method == "POST":
+        file_obj = request.FILES['file']
+        file_name_suffix = file_obj.name.split(".")[-1]
+        if file_name_suffix not in ["jpg", "png", "gif", "jpeg", ]:
+            return JsonResponse({"message": "Wrong file format"})
+
+        path = os.path.join(
+            settings.MEDIA_ROOT,
+            'tinymce',
+        )
+        # If there is no such path, create
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        file_path = os.path.join(path, file_obj.name)
+
+#       file_url = f'{settings.MEDIA_URL}tinymce/{time.time_ns()}_{file_obj.name}'
+        file_url = f'{settings.MEDIA_URL}tinymce/{file_obj.name}'
+
+        if os.path.exists(file_path):
+            return JsonResponse({
+                "message": "file already exist",
+                'location': file_url
+            })
+
+        with open(file_path, 'wb+') as f:
+            for chunk in file_obj.chunks():
+                f.write(chunk)
+
+        return JsonResponse({
+            'message': 'Image uploaded successfully',
+            'location': file_url
+        })
+    return JsonResponse({'detail': "Wrong request"})
```

Comment:
Hi @szabozoltan69

For uploading, we can directly use the storage class instead.
Something like this
```python3
from django.core.files.storage import FileSystemStorage, get_storage_class

DefaultMediaStorageClass = get_storage_class()


def get_full_media_url(media_path):
    if DefaultMediaStorageClass == FileSystemStorage:
        return f"{settings.HTTP_PROTOCOL}://{settings.API_FQDN}{media_path}"
    # For other storage. Assuming they provide absolute URL
    return media_path


def func...():
    ...
    storage = DefaultMediaStorageClass()
    storage.save(file_path, content_file)
    file_url = get_full_media_url(file_path)
```

> WHERE `get_storage_class` provides storage defined using `DEFAULT_FILE_STORAGE`
https://docs.djangoproject.com/en/3.2/ref/files/storage/#django.core.files.storage.get_storage_class

Reply:
Thanks @thenav56

I wonder how content_file variable should have it's content.

Reply:
I get confused with this, but something like this should do :D 
```python
from django.core.files.base import ContentFile

ContentFile(
        file.read().encode('utf-8')
    )

```

Reply:
Well, I could not use this file.read (not obvious which would be the "file"), but maybe file_obj.chunks() could be used like:
```
        with open(file_url, 'wb+') as f:
            for chunk in file_obj.chunks():
                f.write(chunk)
```

Reply:
Maybe `ContentFile(file_obj)` works?
Or directly `storage.save(file_path, file_obj)`